lib/rspec/buildkite/analytics/ci.rb

```diff
@@ -2,30 +2,85 @@
 
 require "securerandom"
 
-module RSpec::Buildkite::Analytics::CI
+class RSpec::Buildkite::Analytics::CI
   def self.env
-    if ENV["BUILDKITE_BUILD_ID"]
-      {
-        "CI" => "buildkite",
-        "key" => ENV["BUILDKITE_BUILD_ID"],
-        "url" => ENV["BUILDKITE_BUILD_URL"],
-        "branch" => ENV["BUILDKITE_BRANCH"],
-        "commit_sha" => ENV["BUILDKITE_COMMIT"],
-        "number" => ENV["BUILDKITE_BUILD_NUMBER"],
-        "job_id" => ENV["BUILDKITE_JOB_ID"],
-        "message" => ENV["BUILDKITE_MESSAGE"],
-        "debug" => ENV["BUILDKITE_ANALYTICS_DEBUG_ENABLED"],
-        "version" => RSpec::Buildkite::Analytics::VERSION,
-        "collector" => RSpec::Buildkite::Analytics::NAME
-      }
-    else
-      {
-        "CI" => nil,
-        "key" => SecureRandom.uuid,
-        "debug" => ENV["BUILDKITE_ANALYTICS_DEBUG_ENABLED"],
-        "version" => RSpec::Buildkite::Analytics::VERSION,
-        "collector" => RSpec::Buildkite::Analytics::NAME
-      }
-    end
+    new.env
+  end
+
+  # The analytics env are more specific than the automatic ci platform env.
+  # If they've been specified we'll assume the user wants to use that value instead.
+  def env
+    ci_env.merge(analytics_env)
+  end
+
+  private
+
+  def ci_env
+    return buildkite if ENV["BUILDKITE_BUILD_ID"]
+    return github_actions if ENV["GITHUB_RUN_NUMBER"]
+    return circleci if ENV["CIRCLE_BUILD_NUM"]
+    return generic if ENV["CI"]
+
+    {
+      "CI" => nil,
+      "key" => SecureRandom.uuid,
+    }
+  end
+
+  def analytics_env
+    {
+      "key" => ENV["BUILDKITE_ANALYTICS_KEY"],
+      "url" => ENV["BUILDKITE_ANALYTICS_URL"],
+      "branch" => ENV["BUILDKITE_ANALYTICS_BRANCH"],
+      "commit_sha" => ENV["BUILDKITE_ANALYTICS_SHA"],
+      "number" => ENV["BUILDKITE_ANALYTICS_NUMBER"],
+      "job_id" => ENV["BUILDKITE_ANALYTICS_JOB_ID"],
+      "message" => ENV["BUILDKITE_ANANLYTICS_MESSAGE"],
+      "debug" => ENV["BUILDKITE_ANALYTICS_DEBUG_ENABLED"],
+      "version" => RSpec::Buildkite::Analytics::VERSION,
+      "collector" => RSpec::Buildkite::Analytics::NAME,
+  }.compact
+  end
+
+  def generic
+    {
+      "CI" => "generic",
```

Comment:
```suggestion
      "CI" => ENV["CI"],
```
Generic providers still provide a value for this key, yes? Should we show whatever their CI name is here?

Reply:
From what I've seen, that value is often "true". Not really the best descriptor for a CI type.

Reply:
What do you mean by "from what I have seen"? Like we don't currently have any customers who just use Test Analytics without Pipelines right? I was assuming that we were going to be launching support for other CI provider with this PR, and that there'd be some instructions like "Set the CI env variable to the name of your CI tool" as that's what all the other values of CI are?

Reply:
Buildkite, GHA and CircleCI all set this to "true". In absence of further research I'm assuming others do as well.

Reply:
Okkkk right so what you mean is that in their docs GHA and CircleCI both specific that they set CI to true and that this is probably a general pattern that other CI providers follow - was not clear that that's where you were looking!